### PR TITLE
Remove DigitalTwins Writable property, rewrite Basic classes to match .NET

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Writable property isn't sent by the service anymore, so we don't need it in our serialization types.

Also adding/changing a couple of Basic types to match what we have in .NET such as BasicDigitalTwinComponentMetadata

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/ComponentSyncSamples.java [hunk 1]
```diff
@@ -85,20 +85,18 @@ public static void runComponentSample() throws JsonProcessingException {
...
                 "Component1",
-                new ModelProperties()
-                    .addProperty("ComponentProp1", "Component value 1")
-                    .addProperty("ComponentProp2", 123)
+                new BasicDigitalTwinComponent()
+                    .addToContents("ComponentProp1", "Component value 1")
+                    .addToContents("ComponentProp2", 123)
             );
 
-        String basicDigitalTwinPayload = mapper.writeValueAsString(basicTwin);
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/ComponentSyncSamples.java:100]: This field wasn't used anywhere, so I deleted it

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/models/BasicDigitalTwinPropertyMetadata.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Contains metadata about changes on properties on a digital twin or component.
+ */
+public class BasicDigitalTwinPropertyMetadata {
+    @JsonProperty(value = DigitalTwinsJsonPropertyNames.METADATA_PROPERTY_LAST_UPDATE_TIME, required = true)
+    private OffsetDateTime lastUpdatedOn;
```
Comment (azabbasi) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/models/BasicDigitalTwinPropertyMetadata.java:15]: I know we decided to use `lastUpdatedOn`, I just wanted to say I think `lastUpdatedTime` might have been a better property name. goodbye

**Review by azabbasi — APPROVED** [review]
Please update the sample/readme file as well. We have a task to make that automated but in the meanwhile we should manually update those snippets.

**Review by drwill-ms — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/models/BasicDigitalTwinPropertyMetadata.java (hunk 2 above)
Comment (timtay-microsoft) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/models/BasicDigitalTwinPropertyMetadata.java:15]: Sadly this is the name the service choose, so I'm not looking to change it.

**Pull request merged**